bin/openaps-packages.sh

```diff
@@ -12,30 +12,35 @@ apt-get install -y sudo
 sudo apt-get update && sudo apt-get -y upgrade
 sudo apt-get install -y git python python-dev software-properties-common python-numpy python-pip watchdog strace tcpdump screen acpid vim locate lm-sensors || die "Couldn't install packages"
 
-# We require jq >= 1.5 for --slurpfile for merging preferences. Debian Jessie ships with 1.4
+# We require jq >= 1.5 for --slurpfile for merging preferences. Debian Jessie ships with 1.4.
 if cat /etc/os-release | grep 'PRETTY_NAME="Debian GNU/Linux 8 (jessie)"' &> /dev/null; then
+   echo "Please consider upgrading your rig to Jubilinux 0.3.0 (Debian Stretch)!"
    sudo apt-get -y -t jessie-backports install jq || die "Couldn't install jq from jessie-backports"
 else
+   # Debian Stretch & Buster ship with jq >= 1.5, so install from apt
    sudo apt-get -y install jq || die "Couldn't install jq"
 fi
 
-# install/upgrade to latest node 8 if neither node 8 nor node 10+ LTS are installed
+# Install/upgrade to latest version of node (v10) using apt if neither node 8 nor node 10+ LTS are installed
 if ! nodejs --version | grep -e 'v8\.' -e 'v1[02468]\.' &> /dev/null ; then
-        # nodesource doesn't support armv6
-        if ! arch | grep -e 'armv6' &> /dev/null ; then
-            sudo bash -c "curl -sL https://deb.nodesource.com/setup_8.x | bash -" || die "Couldn't setup node 8"
-            sudo apt-get install -y nodejs=8.* || die "Couldn't install nodejs"
-        else
-            sudo apt-get install -y nodejs npm || die "Couldn't install nodejs and npm"
-            npm install npm@latest -g || die "Couldn't update npm"
-        fi
-        ## You may also need development tools to build native addons:
-        ##sudo apt-get install gcc g++ make
+   if getent passwd edison; then
+     # Only on the Edison, use nodesource setup script to add nodesource repository to sources.list.d, then install nodejs (npm is a part of the package)
+     curl -sL https://deb.nodesource.com/setup_8.x | bash -
+     sudo apt-get install -y nodejs=8.* || die "Couldn't install nodejs"
+   else
+     sudo apt-get install -y nodejs npm || die "Couldn't install nodejs and npm"
+   fi
+   
+   # Upgrade npm to the latest version using its self-updater
+   sudo npm install npm@latest -g || die "Couldn't update npm"
```

Comment:
What version will we end up with on each platform from this?

Reply:
This should install node v8 on the Edison (both Jessie & Stretch), node v8 on Pi's with Stretch, and node v10 on all Pi's with Buster. The version of npm should be the same across all platforms because it uses the self-updater.